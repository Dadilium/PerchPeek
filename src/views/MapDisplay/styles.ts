import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%'
  },
  landmarksContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    borderTopEndRadius: 20,
    borderTopStartRadius: 20,
    backgroundColor: 'white'
  },
  scrollView: {
    marginVertical: 16,
    flex: 1,
  },
});

export default styles;