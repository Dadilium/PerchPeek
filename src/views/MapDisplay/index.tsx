import React, { useRef, useState } from 'react';
import { View, ScrollView } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { HomeScreenNavigationProp } from '../../AppNavigation';
import { RootState } from '../../store/index';
import LandmarkThumb from '../../components/LandmarkThumb';
import { setLandmarkHeart } from '../../store/Landmarks';
import { RefMap } from '../../components/Map/types';
import Map from '../../components/Map';
import { Landmark } from '../../constants';
import LandmarkThumbStyles from '../../components/LandmarkThumb/styles';
import styles from './styles';

const MapDisplay = ({ navigation }: HomeScreenNavigationProp) => {
  const { mapContainer, landmarksContainer, scrollView } = styles;
  const landmarks: Array<Landmark> = useSelector((state: RootState) => state.landmarks);
  const [dataSourceCords, setDataSourceCords] = useState<Array<number>>([]);
  const [scrollViewRef, setScrollViewRef] = useState<ScrollView | null>();
  const mapRef = useRef<RefMap>();
  const dispatch = useDispatch();

  const onMarkerPressed = (landmark: Landmark) => {
    const { container } = LandmarkThumbStyles;
    const ToX = (landmark.id - 1) * (container.width + container.marginHorizontal);

    scrollViewRef?.scrollTo({
      x: ToX, animated: true,
    });
    mapRef?.current?.changeLandmark(landmark.id);
  };

  return (
    <View style={mapContainer}>
      <Map
        ref={(map: RefMap) => mapRef.current = map}
        landmarks={landmarks}
        onSelection={(landmark) => onMarkerPressed(landmark)}
      />
      <View style={landmarksContainer}>
        <ScrollView
          testID="landmarks-list"
          ref={(ref) => setScrollViewRef(ref)}
          style={scrollView}
          horizontal
        >
          {landmarks.map((landmark, key) => (
            <LandmarkThumb
              key={landmark.id}
              onPress={() => navigation.navigate('Details', { landmark: landmark })}
              onHeartPress={() => dispatch(setLandmarkHeart({ index: key }))}
              onLayout={(x) => {
                dataSourceCords[key] = x;
                setDataSourceCords(dataSourceCords);
              }}
              landmark={landmark}
            />
          ))}
        </ScrollView>
      </View>
    </View>
  );
};

export default MapDisplay;
