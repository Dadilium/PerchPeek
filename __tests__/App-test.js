/**
 * @format
 */

import React from 'react';
import { Provider } from 'react-redux';
import MapDisplay from '../src/views/MapDisplay';
import { render, fireEvent } from '@testing-library/react-native'
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const mockState = {
  landmarks: require('../src/store/data/londonLandmarks.json')
};
const store = mockStore(mockState)

it('all landmarks are on the map', () => {
  const state = store.getState();
  const { getByTestId } = render(
    <Provider store={store}>
      <MapDisplay />
    </Provider>
  );

  state.landmarks.map((landmark) => {
    const marker = getByTestId(`marker-item-${landmark.id}`);
    expect(marker.props.coordinate.latitude).toBe(landmark.latlng.latitude);
    expect(marker.props.coordinate.longitude).toBe(landmark.latlng.longitude);
  });
});

it('the list contains all the landmarks', () => {
  const state = store.getState();
  const { getByTestId, toJSON } = render(
    <Provider store={store}>
      <MapDisplay />
    </Provider>
  );

  expect(toJSON()).toMatchSnapshot();
  state.landmarks.map((landmark) => {
    const item = getByTestId(`landmark-item-label-${landmark.id}`);
    expect(item.children[0]).toBe(landmark.name);
  });
});

it('test marker color change', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MapDisplay />
    </Provider>
  );

  const marker1 = getByTestId(`marker-item-1`);
  const marker2 = getByTestId(`marker-item-2`);

  expect(marker1.props.children[0].props.color).toEqual('grey');
  expect(marker2.props.children[0].props.color).toEqual('grey');
  fireEvent.press(marker1); // change marker1 color to blue
  expect(marker1.props.children[0].props.color).toEqual('#2080c9');
  expect(marker2.props.children[0].props.color).toEqual('grey');
  fireEvent.press(marker2); // change marker1 color back to grey and marker2 to blue
  expect(marker1.props.children[0].props.color).toEqual('grey');
  expect(marker2.props.children[0].props.color).toEqual('#2080c9');
});

it('test hearting landmark on press - mocked store', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MapDisplay />
    </Provider>
  );

  expect(store.getActions().length).toBe(0);
  const item = getByTestId('heartbutton-1');
  fireEvent.press(item); // selecting landmark
  expect(store.getActions().length).toBe(1);
  fireEvent.press(item); // unselecting landmark
  expect(store.getActions().length).toBe(2);
});

// it('test scrollview when marker was pressed', () => {
//   const { getByTestId } = render(
//     <Provider store={store}>
//       <MapDisplay />
//     </Provider>
//   );

//   const scrollview = getByTestId(`landmarks-list`);
//   scrollview.scrollTo = jest.fn();
//   console.log(scrollview)
//   const marker5 = getByTestId(`marker-item-5`);
//   jest.spyOn(ScrollView, 'scrollTo');
//   expect(marker5.props.children[0].props.color).toEqual('grey');
//   fireEvent.press(marker5); // change marker5 color to blue and trigger scrollTo
//   expect(marker5.props.children[0].props.color).toEqual('#2080c9');
//   expect(scrollview.scrollTo).toHaveBeenCalled();

// });

// it('test navigation called', () => {
//   const { getByTestId } = render(
//     <Provider store={store}>
//       <MapDisplay />
//     </Provider>
//   );

//   const item = getByTestId(`landmark-item-1`);
//   item.props.onPress = jest.fn();
//   const spy = jest.spyOn(React, 'onPress')
//   expect(spy).toHaveBeenCalledTimes(0);
//   fireEvent.press(item);
//   // expect(item.props.onPress).toHaveBeenCalledTimes(1);
// })