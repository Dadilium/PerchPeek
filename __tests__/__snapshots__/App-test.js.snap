// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the list contains all the landmarks 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <AIRMap
    handlePanDrag={false}
    initialRegion={
      Object {
        "latitude": 51.500782626551675,
        "latitudeDelta": 0.1,
        "longitude": -0.10652662330828043,
        "longitudeDelta": 0.1,
      }
    }
    onChange={[Function]}
    onLayout={[Function]}
    onMapReady={[Function]}
    onMarkerPress={[Function]}
    region={null}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 51.500782626551675,
          "longitude": -0.12552662330828043,
        }
      }
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      testID="marker-item-1"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 51.50890357879391,
          "longitude": -0.12850203078005293,
        }
      }
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      testID="marker-item-2"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 51.50112240771424,
          "longitude": -0.14135723958435054,
        }
      }
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      testID="marker-item-3"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 51.505737740072995,
          "longitude": -0.07520694901800379,
        }
      }
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      testID="marker-item-4"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 51.518879163719944,
          "longitude": -0.12630217259903156,
        }
      }
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      testID="marker-item-5"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </AIRMapMarker>
  </AIRMap>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopEndRadius": 20,
        "borderTopStartRadius": 20,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      style={
        Object {
          "flex": 1,
          "marginVertical": 16,
        }
      }
      testID="landmarks-list"
    >
      <View>
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="landmark-item-1"
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "borderRadius": 3,
                  "height": 160,
                  "marginHorizontal": 8,
                  "radius": 10,
                  "width": 200,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "flex": 0.8,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://images.pexels.com/photos/460672/pexels-photo-460672.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {
                        "borderRadius": 8,
                      },
                    ]
                  }
                  testID="landmark-item-image-1"
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="heartbutton-1"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "borderRadius": 90,
                          "justifyContent": "center",
                        },
                        Object {
                          "position": "absolute",
                          "right": 10,
                          "top": 8,
                        },
                        Object {
                          "height": 30,
                          "width": 30,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰋕
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "bottom": 8,
                      "color": "white",
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "left": 10,
                      "position": "absolute",
                    }
                  }
                  testID="landmark-item-label-1"
                >
                  Big Ben
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="landmark-item-2"
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "borderRadius": 3,
                  "height": 160,
                  "marginHorizontal": 8,
                  "radius": 10,
                  "width": 200,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "flex": 0.8,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://images.pexels.com/photos/4094000/pexels-photo-4094000.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {
                        "borderRadius": 8,
                      },
                    ]
                  }
                  testID="landmark-item-image-2"
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="heartbutton-2"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "borderRadius": 90,
                          "justifyContent": "center",
                        },
                        Object {
                          "position": "absolute",
                          "right": 10,
                          "top": 8,
                        },
                        Object {
                          "height": 30,
                          "width": 30,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰋕
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "bottom": 8,
                      "color": "white",
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "left": 10,
                      "position": "absolute",
                    }
                  }
                  testID="landmark-item-label-2"
                >
                  National Gallery
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="landmark-item-3"
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "borderRadius": 3,
                  "height": 160,
                  "marginHorizontal": 8,
                  "radius": 10,
                  "width": 200,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "flex": 0.8,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://images.pexels.com/photos/3960662/pexels-photo-3960662.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {
                        "borderRadius": 8,
                      },
                    ]
                  }
                  testID="landmark-item-image-3"
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="heartbutton-3"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "borderRadius": 90,
                          "justifyContent": "center",
                        },
                        Object {
                          "position": "absolute",
                          "right": 10,
                          "top": 8,
                        },
                        Object {
                          "height": 30,
                          "width": 30,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰋕
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "bottom": 8,
                      "color": "white",
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "left": 10,
                      "position": "absolute",
                    }
                  }
                  testID="landmark-item-label-3"
                >
                  Buckingham Palace
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="landmark-item-4"
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "borderRadius": 3,
                  "height": 160,
                  "marginHorizontal": 8,
                  "radius": 10,
                  "width": 200,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "flex": 0.8,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://images.pexels.com/photos/1563230/pexels-photo-1563230.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {
                        "borderRadius": 8,
                      },
                    ]
                  }
                  testID="landmark-item-image-4"
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="heartbutton-4"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "borderRadius": 90,
                          "justifyContent": "center",
                        },
                        Object {
                          "position": "absolute",
                          "right": 10,
                          "top": 8,
                        },
                        Object {
                          "height": 30,
                          "width": 30,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰋕
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "bottom": 8,
                      "color": "white",
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "left": 10,
                      "position": "absolute",
                    }
                  }
                  testID="landmark-item-label-4"
                >
                  Tower Bridge
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="landmark-item-5"
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "borderRadius": 3,
                  "height": 160,
                  "marginHorizontal": 8,
                  "radius": 10,
                  "width": 200,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "flex": 0.8,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://images.pexels.com/photos/2815035/pexels-photo-2815035.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": undefined,
                        "width": undefined,
                      },
                      Object {
                        "borderRadius": 8,
                      },
                    ]
                  }
                  testID="landmark-item-image-5"
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="heartbutton-5"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "white",
                          "borderRadius": 90,
                          "justifyContent": "center",
                        },
                        Object {
                          "position": "absolute",
                          "right": 10,
                          "top": 8,
                        },
                        Object {
                          "height": 30,
                          "width": 30,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰋕
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "bottom": 8,
                      "color": "white",
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "left": 10,
                      "position": "absolute",
                    }
                  }
                  testID="landmark-item-label-5"
                >
                  The British Museum
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
